### ISSUE TITLE
Toolbox button on mobile sometimes overlays clickable items

### ISSUE THREAD

**Issue by sseppi** [noi-techpark/opendatahub-databrowser#590]
The ```+ Toolbox``` button somethimes overlays clickable areas that aren't shiftable, this doesn't allow the user to click on the link/button behind the ```+ Toolbox``` button.

![Image](https://github.com/user-attachments/assets/1cc63a28-632c-49cb-81e2-16a59a2d75c4)

Since the space in the header is little, I don't know if we can move there the button on mobile.
Other tools allow with a long click on the item to move this component that are over other components on another area of the screen.

@pkritzinger @Mazzolintis : do you think could be a solution?

@gappc : how complex will be implement such a functionality?

**Comment by gappc** [comment]
@sseppi I think (hope) the technical complexity is not that much of a problem, but there are some questions and edge-cases that we need to address, like:

- how does the user know that the button can be moved?
- should the button position be remembered?
- what if the user moved the button so close to the border that it's not possible to click / move it again? => is there some way to reset the position? If that is the case, how / where can the user reset the button?
- same argument as before, but different cause: what if the user turns the mobile device by 90 degrees and the button goes out of the visible area?

I'm not sure if the floating button is not always in the way, no matter where the user puts it. Maybe there is some other solution to open the toolbox?

**Comment by sseppi** [comment]
@gappc  thank you for your reply.

> * how does the user know that the button can be moved?

Other tools does the same on mobile. When I had the problem, the first thing I tried was to long click on the button and try to move it in another place.

> * should the button position be remembered?

Ideally yes in the session cookies.

> * what if the user moved the button so close to the border that it's not possible to click / move it again? => is there some way to reset the position? If that is the case, how / where can the user reset the button?

The area where the button can be moved should be limited only to the visible and clickable area. In by case the user can reset the position by clearing the cookie.

> * same argument as before, but different cause: what if the user turns the mobile device by 90 degrees and the button goes out of the visible area?

The position should be relative and not absolute. In that way, by rotating the device, the button will be replaced always in a visible area. If by rotating the device the button overlays some clickable area or important data, the user has always the possibility to move it.
 
> I'm not sure if the floating button is not always in the way, no matter where the user puts it. Maybe there is some other solution to open the toolbox?

I'm open also to other solution and positioning of the open toolbox functionality on mobile. The important thing is that the user doesn't loose functionality.

**Comment by pkritzinger** [comment]
@gappc @sseppi let's deepdive on this in a call maybe.
I like Stefanos idea but see the problems posted by Christian.

Please consider that floating Buttons are also a standard defined by Google. 
https://m3.material.io/components/floating-action-button/overview

Maybe a simple solution could be to optimize the size and positioning of our floating button?

**Comment by pkritzinger** [comment]
@sseppi here's our two proposals:
![Image](https://github.com/user-attachments/assets/895b1008-0fe9-4201-88e3-1b433867289a)

- We reduced the size of the buttons and changed the text from horizontal to vertical to make sure that it less conflicts with the other interaction elements.
- We should also foresee more margin at the bottom of each screen so that we assure that the last field is not hidden bei the toolbox.
![Image](https://github.com/user-attachments/assets/43617ad0-b413-4753-9e6c-36a68c14e1e5)

**Comment by Xeroxi91** [comment]
comment

**Comment by sseppi** [comment]
@Xeroxi91 I noticed a strange behaviour of the toolbox button on mobile that becomes 100% high in both table and detail view. As you can see from the screenshot below, in the detail vew also the Toolbox text is missing.

![Image](https://github.com/user-attachments/assets/b2886a99-a19b-4730-ad4a-1de66cccdf44)
![Image](https://github.com/user-attachments/assets/fd46027f-79f4-4f0e-a612-3c8b65884021)

Can you please fix this issue as soon as possible?

**Comment by gappc** [comment]
@sseppi @Xeroxi91 the issue seems to be related to my change on button styles over here: https://github.com/noi-techpark/it.bz.opendatahub.databrowser/commit/5c446a1868a792d9f9c4b6a0ce1cd252ca505d66#diff-13f5c23137eadfa3f543d7a4db25acd6e8aba085b14972976dd384cac4d95a62R67

The `relative` class was added as part of the feature "Extend button component with an indicator", a requirement that we have for the map (see image below)

![Image](https://github.com/user-attachments/assets/bbb8c8c3-e27f-42d5-b8f5-61c25d110262)

@Xeroxi91 when you fix the issue, please make sure that the indicators still work, thx 👍

**Comment by Xeroxi91** [comment]
@gappc the `relative` class is necessary for the select?

**Comment by gappc** [comment]
@Xeroxi91 no, it allows to position `absolute` the indicator (green circle) inside the button

**Comment by Xeroxi91** [comment]
@gappc Ok, I've verified that removing the class from the `ButtomCustom` component's [style.ts](https://github.com/noi-techpark/it.bz.opendatahub.databrowser/blob/3156f29df470dd772e917cf7b9205e22b6319d37/databrowser/src/components/button/styles.ts) file resolves the issue. Additionally, it's necessary to add the `relative` class to the `ButtonCustom` component within the [MapViewHeader](https://github.com/noi-techpark/it.bz.opendatahub.databrowser/blob/3156f29df470dd772e917cf7b9205e22b6319d37/databrowser/src/domain/datasets/ui/mapView/MapViewHeader.vue) component specifically for mobile views. I've already tested this locally, and it works correctly. Can I proceed with the pull request?

**Comment by gappc** [comment]
@Xeroxi91 would it be possible to keep support for the indicator in the `ButtomCustom` component? We need the indicator at least in two places (maybe more in the future), it would be nice if we don't have to manually apply the `relative` class when necessary.

Given that the `ButtomCustom` component already supports the `indicator` property, please check if you can apply the `relative` class depending on that property.

**Comment by Xeroxi91** [comment]
Yes, this can be done. I'll update the `ButtonCustom` component to automatically apply the `relative` class based on the existing `indicator` property, so we won't need to manually handle it in each usage.
